packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -1112,8 +1117,18 @@ class HotRunner extends ResidentRunner {
   }
 }
 
+/// The [ProjectFileInvalidator] track the dependencies for a running
+/// application to determine when they are dirty.
 class ProjectFileInvalidator {
-  const ProjectFileInvalidator();
+  ProjectFileInvalidator(
+    this._fileSystem,
```

Comment:
I tend to prefer named parameters when a method/constructor involves more than one parameter.

Reply:
Seems like a good idea. Generally there is a bit more boilerplate involved if we want to use named parameters but still have private fields which is why I avoided it.

Reply:
fair :)

Reply:
Still updated to use named parameters & private fields, its really a minor complaint. Plus it reads a lot better